clang/lib/AST/Interp/ByteCodeExprGen.cpp

```diff
@@ -1191,6 +1202,25 @@ bool ByteCodeExprGen<Emitter>::VisitConstantExpr(const ConstantExpr *E) {
   return this->delegate(E->getSubExpr());
 }
 
+template <class Emitter>
+bool ByteCodeExprGen<Emitter>::VisitPPEmbedExpr(const PPEmbedExpr *E) {
+  for (const IntegerLiteral *IL : E->underlying_data_elements()) {
+    if (!this->visit(IL))
+      return false;
+  }
```

Comment:
So this will potentially push N integers on the stack... who will remove them again? What't the usual AST structure a `PPEmbedExpr` can be found in?

Reply:
Yeah, I think this implementation doesn't make much sense since with current logic any of evaluators or codegen shouldn't see `PPEmbedExpr`, since it is expected to be broken down into `EmbedSubscriptExpr`s or pure `IntegerLiteral`s. I may have added it before I realized that there is the stack. I'll remove it.

Reply:
Yeah ok, removing it results in failure in one of the tests:
```
static_assert(                         
#embed <jk.txt> limit(1) suffix(== 'j')
);                                     
```
Which is basically a stray `#embed` outside of an initializer so it is not transformed in any way. However it only works due to `limit(1)`, once I remove that we get non-empty stack assertion which is expected. I'll try to figure out how to handle stray `#embed`s appearing in `static_assert`...

Reply:
You should also test other scenarios where embed can be used outside of an initializer, such as:
```
constexpr int func(int i, int) { return i; }
static_assert(
  func(
#embed <jk.txt>
  ) == 'j');

template <int N>
struct S {};

S<
#embed <jk.txt> limit(1)
> s;
```
it's fine for these to go slow or to diagnose as "sorry, not supported, stop being so clever!", it's mostly to ensure we don't assert/crash.